client/src/controllers/ActionController.test.js

```diff
@@ -199,4 +199,61 @@ describe('ActionController', () => {
       expect(textarea.selectionEnd).toBe(textarea.value.length);
     });
   });
+
+  describe('reset method', () => {
+    beforeEach(async () => {
+      await setup(
+        `<input
+          id="reset-test"
+          value="the default"
+          type="text"
+          data-controller="w-action"
+          data-action="some-event->w-action#reset"
+        />`,
+      );
+    });
+    it('should not update value when event is null', () => {
+      const input = document.getElementById('reset-test');
+
+      input.dispatchEvent(new CustomEvent(''));
+
+      expect(input.value).toBe('the default');
+      expect(input.value).not.toBe('not the default');
+    });
+
+    it('should not change value when event does not have required properties', () => {
+      const input = document.getElementById('reset-test');
+
+      input.dispatchEvent(
+        new CustomEvent('some-event', {
+          detail: { wrongValue: 'the default' },
+        }),
+      );
+
+      expect(input.value).toBe('the default');
+      expect(input.value).not.toBe('not the default');
+    });
+
+    it('should not change value when current value and new value are the same', () => {
+      const input = document.getElementById('reset-test');
+
+      input.dispatchEvent(
+        new CustomEvent('some-event', { detail: { value: 'the default' } }),
+      );
+
+      expect(input.value).toBe('the default');
+      expect(input.value).not.toBe('not the default');
+    });
+
+    it('should change value when existing value and new value are different', () => {
+      const input = document.getElementById('reset-test');
+
```

Comment:
This test should add a line where the value is set to something other than the default. 

Maybe add a line that sets `input.value = 'not the defaul '` first. 

Remember the test should validate our desired behaviour. 

- Input enters the DOM with some value preset in the value attribute. 
- user changes that value manually
- event listener (data-action) gets triggered
- the reset method gets called
- the manually changes value should be reset to the value in the DOM originally